e2e_tests/utils/assertions.py

```diff
@@ -0,0 +1,68 @@
+import warnings
+
+from axe_playwright_python.sync_playwright import Axe
+
+axe = Axe()
+
+
+class AccessibilityWarning(UserWarning):
+    pass
+
+
+def generate_accessibility_report(violations):
+    return "\n".join(format_violation(v) for v in violations)
+
+
+def assert_critical_violations(violations, page_url):
+    critical_or_serious_violations = []
+
+    for violation in violations:
+        if violation["impact"] in ["critical", "serious"]:
+            critical_or_serious_violations.append(violation)
+
+    if critical_or_serious_violations:
+        report = generate_accessibility_report(critical_or_serious_violations)
+        # TODO: Once we have fixed all our critical/serious accessibility violations, make this fail the test
+        warnings.warn(
+            f"\nCritical/Serious Accessibility Violations Detected ({page_url}):\n{report}", AccessibilityWarning
+        )
+
+
+def assert_other_violations(violations, page_url):
```

Comment:
Is assert sensible here in the name when we're not intending to necessarily raise an assertion for non-critical/serious violations? Would something more neutral like `check` make sense?

Reply:
Yep, I agree 👍 will update